cpp/oneapi/dal/table/backend/interop/host_soa_table_adapter.hpp

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include <daal/include/data_management/data/soa_numeric_table.h>
+
+#include "oneapi/dal/table/homogen.hpp"
+#include "oneapi/dal/table/row_accessor.hpp"
+#include "oneapi/dal/table/column_accessor.hpp"
+#include "oneapi/dal/backend/interop/error_converter.hpp"
+#include "oneapi/dal/backend/interop/daal_object_owner.hpp"
+#include "oneapi/dal/table/backend/interop/block_info.hpp"
+
+namespace oneapi::dal::backend::interop {
+
+// This class shall be used only to represent immutable data on DAAL side. Any
+// attempts to change the data inside objects of that class lead to undefined
+// behavior.
+template <typename Data>
```

Comment:
Why template parameter is needed here?

Reply:
SOA table may hold heterogeneous data by design

Reply:
applied